core/src/repair_service.rs

```diff
@@ -197,14 +239,26 @@ impl RepairService {
         Ok(repairs)
     }
 
+    #[cfg(test)]
+    fn test_generate_repairs(
+        repairs: Vec<RepairType>,
+        blockstore: &Blockstore,
+        root: Slot,
+        max_repairs: usize,
+    ) -> Result<Vec<RepairType>> {
+        let incomplete = Self::find_incomplete_slots(&Pubkey::default(), blockstore, root);
+        Self::generate_repairs(repairs, blockstore, root, &incomplete, max_repairs)
+    }
+
     fn generate_repairs(
+        mut repairs: Vec<RepairType>,
```

Comment:
can just use &mut instead of returning a Vec<>

Reply:
rust in theory will optimize out the allocation because the ownership is within calls in this file